src/mongo_client.ts

```diff
@@ -32,6 +32,21 @@ import type { SrvPoller } from './sdam/srv_polling';
 import type { Connection } from './cmap/connection';
 import type { LEGAL_TLS_SOCKET_OPTIONS, LEGAL_TCP_SOCKET_OPTIONS } from './cmap/connect';
 
+/** @public */
+export const ServerApiVersion = {
+  v1: '1'
+};
+
+/** @public */
+export type ServerApiVersionId = typeof ServerApiVersion[keyof typeof ServerApiVersion];
+
+/** @public */
+export interface ServerApi {
+  version: string | ServerApiVersionId;
```

Comment:
Instead of freezing the `ServerApi` on the getter in `MongoClient`, could we instead just define the properties as `readonly` here? Or does that not give us enough protection somehow?

Reply:
I think `readonly` only applies at the level of the TypeScript compiler, so someone using plain JS would be able mutate the property. I'd be interested to hear your thoughts on this @nbbeeken

Reply:
Indeed the `readonly` modifier only effects type checking. I think freezing is worth it here, since it is a construction time only setting, we can just jump ahead of any bugs trying to change serverApi at a later time by doing this.

Reply:
Ok great, thanks!